keras_cv/visualization/plot_segmentation_mask_gallery.py

```diff
@@ -0,0 +1,123 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+
+from keras_cv import utils
+from keras_cv.utils import assert_matplotlib_installed
+from keras_cv.visualization.plot_image_gallery import plot_image_gallery
+
+
+def reshape_masks(segmentation_masks):
+    rank = len(segmentation_masks.shape)
+    if rank == 3:
+        # (B, H, W)
+        return segmentation_masks[..., np.newaxis]
+    elif rank == 4:
+        # (B, H, W, num_channels) OR (B, H, W, 1)
+        if segmentation_masks.shape[-1] == 1:
+            # Repeat the masks 3 times in order to build 3 channel
+            # segmentation masks.
+            return segmentation_masks.repeat(repeats=3, axis=-1)
+        else:
+            return np.argmax(segmentation_masks, axis=-1).repeat(
+                repeats=3, axis=-1
+            )
+
+
+def plot_segmentation_mask_gallery(
+    images,
+    value_range,
+    num_classes,
+    y_true=None,
+    y_pred=None,
+    rows=3,
+    cols=3,
+    **kwargs
+):
+    """Plots a gallery of images with corresponding segmentation masks.
+
+    Args:
+        images: a Tensor or NumPy array containing images to show in the
+            gallery. The images should be batched and of shape (B, H, W, C).
+        value_range: value range of the images. Common examples include
+            `(0, 255)` and `(0, 1)`.
+        num_classes: number of segmentation classes.
+        y_true: (Optional) a Tensor or NumPy array representing the ground truth
+            segmentation masks. The ground truth segmentation maps should be
+            batched.
+        y_pred: (Optional)  a Tensor or NumPy array representing the predicted
+            segmentation masks. The predicted segmentation masks should be
+            batched.
+        kwargs: keyword arguments to propagate to
+            `keras_cv.visualization.plot_image_gallery()`.
+
+    Usage:
+    ```python
+    train_ds = tfds.load(
+        "oxford_iiit_pet", split="train", with_info=False, shuffle_files=True
+    )
+
+    def unpackage_tfds_inputs(inputs):
+        image = inputs["image"]
+        segmentation_mask = inputs["segmentation_mask"]
+        return image, segmentation_mask
+
+    train_ds = train_ds.map(unpackage_tfds_inputs).ragged_batch(16)
+    images, segmentation_masks = next(iter(train_ds.take(1)))
+
+    keras_cv.visualization.plot_segmentation_mask_gallery(
+        images,
+        value_range=(0, 255),
+        num_classes=3, # The number of classes for the oxford iiit pet dataset
+        y_true=segmentation_masks,
+        y_pred=None,
+        scale=3,
+        rows=2,
+        cols=2,
+    )
+    ```
+
+    ![Example segmentation mask gallery](https://i.imgur.com/aRkmJ1Q.png)
+    """
+    assert_matplotlib_installed("plot_segmentation_mask_gallery")
+
+    plotted_images = utils.to_numpy(images)
+
+    plotted_y_true = utils.to_numpy(y_true)
```

Comment:
This makes me think `y_true` is required -- should we wrap it in an `if y_true is not None:` (like we do for `y_pred`)

Reply:
My rationale here is that if we do not keep `y_true` required this API is as good as `plot_image_gallery`. Does that not make the usage redundant?

Reply:
That makes sense to me!